src/components/CardList.jsx

```diff
@@ -1,5 +1,45 @@
-const CardList = () => {
-  return ();
+import Card from "./Card";
+import Button from "./Button";
+import Search from "./Search";
+import React, { useState, useEffect } from "react";
+const CardList = ({data}) => {
+  const limit = 10;
+  const defaultDataset = data.slice(0, limit);
+  const [offset, setOffset] = useState(0);
+  const [products, setProducts] = useState(defaultDataset);
+  const handlePrevious = () => {
+    setOffset(offset - limit);
+  }
+  const handleNext = () => {
+    setOffset(offset + limit);
+  }
+  useEffect(() => {
+    setProducts(data.slice(offset, offset + limit));
+  }, [offset, limit, data]);
+  const filterTags = (tagQuery) => {
+    const filtered = data.filter(product=>{
+      if (!tagQuery) {
+        return product
+      }
+      return product.tags.find (({title}) => title===tagQuery)
```

Comment:
This will evaluate, but its good habit to have spacing. 

```suggestion
      return product.tags.find (({title}) => title === tagQuery)
```